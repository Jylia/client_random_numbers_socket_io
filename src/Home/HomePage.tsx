import * as React from 'react';
import { History } from 'history';
import { Header, Button, Icon } from 'semantic-ui-react';
import { MobileContext } from '../Layout/MobileContext';

interface IHomePageProps {
    history: History;
    isMobile: boolean;
}

interface IHomePageState {}

class HomePage extends React.Component<IHomePageProps, IHomePageState> {
    
    render() {
        return (
            <MobileContext.Consumer>
                {({isMobile}) => (
                    <React.Fragment>
                        <Header
                            as='h1'
                            content='Random number charts'
                            inverted
                            style={{
                                fontSize: isMobile ? '2em' : '4em',
                                fontWeight: 'normal',
                                marginBottom: 0,
                                marginTop: isMobile ? '1.5em' : '3em',
                            }}
                        />
                        <Header
                            as='h2'
                            content='Look at changes in real-time'
                            inverted
                            style={{
                                fontSize: isMobile ? '1.5em' : '1.7em',
                                fontWeight: 'normal',
                                marginTop: isMobile ? '0.5em' : '1.5em',
                            }}
                        />
                    </React.Fragment>
                )}
            </MobileContext.Consumer>
        );
    }
}

export default HomePage;