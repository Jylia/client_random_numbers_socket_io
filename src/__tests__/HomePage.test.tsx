import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { history } from '../History';
import HomePage from '../Home/HomePage';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<HomePage history={history} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
    const tree = renderer
        .create(<HomePage history={history} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});