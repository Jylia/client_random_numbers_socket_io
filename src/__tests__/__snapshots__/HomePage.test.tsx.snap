// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h1
    className="ui inverted header"
    style={
      Object {
        "fontSize": "4em",
        "fontWeight": "normal",
        "marginBottom": 0,
        "marginTop": "3em",
      }
    }
  >
    Random number charts
  </h1>,
  <h2
    className="ui inverted header"
    style={
      Object {
        "fontSize": "1.7em",
        "fontWeight": "normal",
        "marginTop": "1.5em",
      }
    }
  >
    Look at changes in real-time
  </h2>,
]
`;
